packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -171,7 +177,11 @@ function act(callback: () => Thenable) {
 
     // flush effects until none remain, and cleanup
     try {
-      flushWork();
+      if (actingUpdatesScopeDepth === 1) {
```

Comment:
Nit:
```suggestion
      if (actingUpdatesScopeDepth <= 1) {
```

Reply:
really? it shouldn't do that if it's 0 (and it ideally should never be 0 here). going to push back on this.

Reply:
tbh, it should probably throw an error if it ever gets there